examples/extensions/r2/hello_r2.py

```diff
@@ -18,13 +19,27 @@ def func(ql: Qiling, *args, **kwargs):
 def my_sandbox(path, rootfs):
     ql = Qiling(path, rootfs, verbose=QL_VERBOSE.DEFAULT)
     r2 = R2(ql)
-    assert(ql.loader.images[0].base == r2.baddr)
-    addrs = ql.mem.search(b'Hello world!')
-    addr = r2.strings['Hello world!'].vaddr
-    assert(addr == addrs[0])
+
+    # search bytes sequence using ql.mem.search
+    addrs = ql.mem.search(b'Hello worl')  # return all matching results
+    # search string using r2
+    addr = r2.strings['Hello world!'].vaddr  # key must be exactly same
+    print(r2.strings['Hello world!'].__class__)
+    # write to string using ql.mem.write
     ql.mem.write(addr, b"No hello, Bye!\x00")
+
+    # get function address and hook it
     ql.hook_address(func, r2.functions['main'].offset)
     ql.run()
 
 if __name__ == "__main__":
     my_sandbox(["rootfs/x86_windows/bin/x86_hello.exe"], "rootfs/x86_windows")
+
+    # test shellcode mode
+    ARM64_LIN = unhexlify('420002ca210080d2400080d2c81880d2010000d4e60300aa01020010020280d2681980d2010000d4410080d2420002cae00306aa080380d2010000d4210400f165ffff54e0000010420002ca210001caa81b80d2010000d4020004d27f0000012f62696e2f736800')
```

Comment:
the `binascii` module is obsolete; you can use `bytes.fromhex` directly without the need of an extra module.

Reply:
Just copy it from our example `shellcode_run.py`, will change it.